### ISSUE TITLE
Incorrect backtesting period

### ISSUE THREAD

**Issue by noudcorten** [dema-trading-ai/engine#41]
The current way of fetching the ohlcv data can be observed in the following screenshot:
![image](https://user-images.githubusercontent.com/44836798/109291391-36019e00-7829-11eb-81a0-2e7cc24a1293.png)

But this implementation is wrong. When looking at the documentation for binance.fetch_ohlcv() (line 1364: https://github.com/ccxt/ccxt/blob/b59bb1ce93bde9832cdb22d4a85cc7a77abc9672/python/ccxt/binance.py#L29) there is a parameter called 'limit' which is currently not used. A quick intuition behind this paramter is that 'limit' defines how many candles you want to fetch. So if timeframe='5m' and limit=12 it will fetch 12 candles from the specified date using 5m timeframe (covering 12*5m=60m=1h).

The problem with our current implementation is that the 'limit' parameter is not defined, meaning the function will choose this parameter itself. After some testing I found that not specyfying the limit paramater results in 'limit=500'. Furthermore, the maximum amount of candles you are able to fetch is 'limit=1000'.

Because the fetch_ohlcv() will always fetch 500 candles from the given date, it can greatly surpass our defined 'testing_to'. This means that our fetched data could be far beyond what we want. This needs to be fixed.

**Comment by WillemGerritzen** [comment]
So the idea would be to compute the limit based on the from and to dates and pass that on as the 'limit=' argument to self.exchange.fetch_ohlcv? 

As an aside, do you know what happens to the extra data that's fetched?

**Comment by noudcorten** [comment]
@WillemGerritzen 
>So the idea would be to compute the limit based on the from and to dates and pass that on as the 'limit=' argument to self.exchange.fetch_ohlcv?
>
Yes that is the idea. But since the maximum amount for this variable is 'limit=1000' we have to do this in a loop. I currently implemented the following in PR #43:
![image](https://user-images.githubusercontent.com/44836798/109394677-7b58c500-7928-11eb-8842-0c45e9cbd134.png)
Here, in every loop we calculate how many candles still need to be fetched (line 131) and then try to fetch these. And since the maximum amount is 1000, we always set 'limit' to 1000 unless we need less than this amount. Finally, we set the new starting date according to how many candles we have fetched (line 139) and we repeat the proces.

> As an aside, do you know what happens to the extra data that's fetched?
>
The extra data that is fetched is currently included for backtesting. So this means that if we want data between 2020-01-01 to 2020-02-01 it could be the case we fetch more than needed. Meaning that we for example fetch data from 2020-01-01 to 2020-02-04 (just a random example). Meaning our backtest uses more data than wnated.
